src/main/java/org/pawaneuler/Generator/TransactionLogGenerator/RandomGenerator.java

```diff
@@ -28,16 +27,17 @@ public RandomGenerator() {
      */
     public RandomGenerator(int maxVariety) {
         this.randomizer = new Random();
-        this.productVariety = randomProducts((maxVariety > 26) ? 26 : maxVariety);
-        this.maxVariety = productVariety.size();
+        this.maxVariety = (maxVariety > 26) ? 26 : maxVariety;
+        this.productVariety = getStrings(products, this.maxVariety);
+        
         
     }
 
     public int getMaxVariety() {
         return maxVariety;
```

Comment:
Just to make the source of it more obvious.

```suggestion
        return this.maxVariety;
```